sdk/identity/identity/src/credentials/visualStudioCodeCredential.ts

```diff
@@ -88,14 +98,26 @@ export interface VisualStudioCodeCredentialOptions extends TokenCredentialOption
 export class VisualStudioCodeCredential implements TokenCredential {
   private identityClient: IdentityClient;
   private tenantId: string;
+  private cloudName: VSCodeCloudNames;
 
   /**
    * Creates an instance of VisualStudioCodeCredential to use for automatically authenticating via VSCode.
    *
    * @param options Options for configuring the client which makes the authentication request.
    */
   constructor(options?: VisualStudioCodeCredentialOptions) {
-    this.identityClient = new IdentityClient(options);
+    // We want to make sure we use the one assigned by the user on the VSCode settings.
+    // Or just `AzureCloud` by default.
+    this.cloudName = (getPropertyFromVSCode("azure.cloud") || "AzureCloud") as VSCodeCloudNames;
+
+    // Picking an authority host based on the cloud name.
+    const authorityHost = mapVSCodeAuthorityHosts[this.cloudName];
+
+    this.identityClient = new IdentityClient({
```

Comment:
The authority host specified via options passed in code, should get the higher priority over the cloudName.
Is that the case currently  ?

Reply:
I had it as the opposite! I'll flip the two lines below.

Reply:
I did the deed!